lib/scraper_test.rb

```diff
@@ -30,17 +31,41 @@ def install_tasks
           it 'should contain the expected data' do
             # TODO: Make this configurable
             Dir['test/data/*.yml'].each do |file|
-              yaml_data = YAML.load_file(file).to_h
-              url = yaml_data[:url]
-              class_to_test = Object.const_get(yaml_data[:class])
-              VCR.use_cassette(File.basename(url)) do
-                response = class_to_test.new(response: Scraped::Request.new(url: url).response)
-                response.to_h.must_equal yaml_data[:to_h]
+              VCR.use_cassette(File.basename(file)) do
+                test = Instructions.new(file)
+                response = test.class_to_test.new(response: Scraped::Request.new(url: test.url).response)
+                response.to_h.must_equal test.expected
               end
             end
           end
         end
       end
     end
   end
+
+  class Instructions
+    def initialize(file)
+      @file = file
+    end
+
+    def url
+      yaml_data[:url] or raise "No url provided by #{File.basename(file)}."
```

Comment:
You're calling `File.basename(file)` in three places, which is a good sign that you're missing a method here somewhere. However, rather than creating such a method yourself, I'd suggest upgrading the filename to a `Pathname`, and then you can just call `file.basename` on it.